src/overlays/actors/ovl_Bg_Ingate/z_bg_ingate.c

```diff
@@ -30,17 +31,93 @@ const ActorInit Bg_Ingate_InitVars = {
     (ActorFunc)BgIngate_Update,
     (ActorFunc)BgIngate_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Ingate/BgIngate_SetupAction.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Ingate/BgIngate_Init.s")
+extern UNK_TYPE D_060011B8;
+extern Gfx D_06001040[];
+
+void BgIngate_SetupAction(BgIngate* this, BgIngateActionFunc func) {
+    this->actionFunc = func;
+}
+
+void BgIngate_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgIngate* this = THIS;
+    s32 pad;
+    s32 sp32 = 0;
+
+    DynaPolyInfo_SetActorMove(&this->dyna, 0);
+    DynaPolyInfo_Alloc(&D_060011B8, &sp32);
+
+    this->dyna.dynaPolyId = DynaPolyInfo_RegisterActor(globalCtx, &globalCtx->colCtx.dyna, &this->dyna.actor, sp32);
+
+    if ((globalCtx->sceneNum != 0x63 || gSaveContext.linkAge != 0) ||
+        (((gSaveContext.eventChkInf[1] & 0x100) != 0) && (gSaveContext.cutsceneIndex != 0xFFF0))) {
+        Actor_Kill(&this->dyna.actor);
+        return;
+    }
+
+    Actor_SetScale(&this->dyna.actor, 0.1f);
+    if (((this->dyna.actor.params & 1) != 0) && ((gSaveContext.eventInf[0] & 0xF) == 6)) {
+        globalCtx->csCtx.frames = (u16)0U;
+        BgIngate_SetupAction(this, &func_80892890);
+        return;
+    }
+    BgIngate_SetupAction(this, &func_80892990);
+}
+
+void BgIngate_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgIngate* this = THIS;
+
+    DynaPolyInfo_Free(globalCtx, &globalCtx->colCtx.dyna, (s32)this->dyna.dynaPolyId);
+}
+
+void func_80892890(BgIngate* this, GlobalContext* globalCtx) {
+    s32 phi0;
+    s16 phi1;
+    s16 ctxFrames;
+
+    if (globalCtx->csCtx.frames >= 0x32) {
```

Comment:
globalCtx->csCtx.frames is a frame counter for a cutscene. so all references to it make more sense in decimal.
similarly, the temp variable name should probably be csFrames, since ctx doesnt really tell you much about what it actually is